### DISCUSSION TITLE
Jitter buffer indicator going crazy on all my sound cards on a fresh windows 10 install

### DISCUSSION THREAD

**Discussion by newlaurent62** [jamulussoftware/jamulus#1494] [Technical Support & Help]
**Describe the bug**
I made a fresh install of Windows 10.

I started a Jamulus server and connected to it using Jamulus client on the same machine.
The jitter buffer changed from green to red and red to green very often on all my sound cards (Zoom U22 and Berhinger XR18) even if i set local and server jitter buffer to 20 !

I installed the last ASIO drivers for all my sound cards as well as ASIO4ALL.

It was working on my previous windows 10 installation.

**Expected behavior**
The  jitter buffer indicator should stabilize to green. These ASIO sound cards worked on my previous windows 10 installation.

**Operating system**
Windows 10 version : 19042.867, Windows Feature Experience Pack 120.2212.551.0 .

**Version of Jamulus**
3.7.0

**Notes**
The previous windows 10 installation was made by the maker. I did the new one so may be i forgot some steps to get my windows ready for Jamulus ? (drivers ...)

**Comment by hoffie** [comment]
Thanks for your report. This looks like a machine-specific issue at first glance, therefore I'm moving this to a discussion. If it turns out to be a Jamulus bug, we'll open a more specific Issue later.

**Comment by hoffie** [comment]
If I understand correctly, you are running both a Jamulus server and a client on the same machine, are connecting directly (i.e. 127.0.0.1 or localhost) and you are still seeing Jitter buffer issues?

This should not happen. I wonder if this could be related to lack of CPU resources? I'm not a Windows expert, so I hope others can chime in.

**Reply by newlaurent62** [reply]
Yes i am connected to localhost (127.0.0.1).

My computer is an i7 and can handle Jamulus Server as well as at least one Jamulus Client and other stuff. I already did it successfully on Windows 10 on the same machine and often do it on the same machine with Linux and a lot more softwares.

**Reply by newlaurent62** [reply]
My CPU is ok about 20%.

**Reply by newlaurent62** [reply]
When i connect to a distant server the jitter buffer indicator stabilizes. As soon as i connect to localhost the jitter buffer is crazy ...

**Reply by newlaurent62** [reply]
And my friend has the same behavior (and a different sound card ...) on windows 10.

**Comment by dcorson-ticino-com** [comment]
Have you recently had an Windows update? 
I am running  V1909 build 18363.1440.
I can run server and client on the same machine without any problems.

**Reply by newlaurent62** [reply]
I downloaded the latest windows install and install it from scratch. My windows version is indicated in my original message (it seems newer than yours).

**Reply by newlaurent62** [reply]
This behavior may be related to a windows update. You may be right.

**Reply by newlaurent62** [reply]
I installed all windows update that were available.

See also this discussion on facebook  (Jamulus official group):
https://www.facebook.com/groups/619274602254947/permalink/887517375430667

Some people say that the problem described here is normal for Windows users. 

But on Linux, i do not have this behavior even with exactly the same hardware.

So i'm confused ...

**Reply by newlaurent62** [reply]
My Windows 10 version is now : 19042.906 and i still have the same behavior (jitter buffer indicator  going crazy when connecting to localhost but stays green when connecting to distant server on the internet).

**Reply by dcorson-ticino-com** [reply]
I'm confused too.
I tried 3 configurations today.
1- server and client on same computer, connect with 127.0.0.1, ping 0
2- server and client on same local network, ping 0
3- server on cloud, ping 12
I see no difference in the jitter buffer behaviour.
The servers are all 3.7.0 as is the client.

Maybe this is coming from background tasks running on your compi.
I would try eliminating all the background things and programs called at startup to see if that makes a difference.

**Reply by newlaurent62** [reply]
I have only Jamulus Server / Jamulus client and system running. I have even uninstall the anitvirus to test but still the same.

I am running Windows 20H2 up to date. This might be the cause of my "special" behavior if you can't reproduce it on your setup which is not the same as me.

**Comment by ghost** [comment]
Do your server options use -T ? ( Jamulus -s -T || Jamulus -s ) Does one work better?  (wild guessing)

**Reply by newlaurent62** [reply]
It's a good idea, i think i already checked the multithreading option. But i will do it again.

**Reply by newlaurent62** [reply]
Ok nice ! The multithreading has no effect on my behavior but fastupdate yes !

As soon as i removed the --fastupdate option, the indicator stays green ...

**Reply by newlaurent62** [reply]
I also disabled "Enable small Network buffers" !

**Reply by newlaurent62** [reply]
Thanks for your help !

**Comment by newlaurent62** [comment]
I have tested my two sound cards one with 128 and another one with 64 audio buffers with fastupdate on the server and with small network buffers enabled, the jitter buffer has become crazy ... 

I noticed that removing fastupdate option from the server changing the audio buffers to 128 and disabling "Enable small Network Buffers" stabilizes the jitter buffer indicator to green.

Do i have to consider that Windows cannot managed these configuration compared to Linux that handle these situations very well on the same sound cards and machine ? Or is it possible that it is a bug on Windows ? Or do i forgot a tweak to optimize my Windows performances ?

**Reply by andresteinert** [reply]
I experienced the same when I tried to set up an ideal environment for testing purposes. For me, it seemed like I couldn't optimize my sporadic dropouts on the remote Linux server any further, that's why I wanted to check it in my LAN environment. The easiest was using another Windows PC being the server as it was already available.

I didn't run it localhost but experienced the same problem with -F option on Windows server side (with enabled small network buffers on the Jamulus client). I didn't run 3.7.0 of Jamulus like you, but 3.8.0rc1 (server and client).

When I switched to a local Linux server (running Jamulus with the same -F option), the problem disappeared.

I'm wondering if there is a (way smaller) problem inherent in the Windows Jamulus client too because I'm experiencing rare dropouts even in this ideal local environment without packet losses and guaranteed lowest ping times. Maybe I'll try building a local Linux PC as a client to be able to say if there is maybe a small problem/bug causing these rare dropouts (which add to the normal network dropouts when using a remote server).

**Reply by 38github** [reply]
My experience is also that the jitter is extremally better in the Linux distributions I have tried it on compared to Windows with the exact computer and devices. I have no evidence or data for this but my server on BSD is fantastically stable and the auto jitter almost immediatelly can go down to 2/2.

**Reply by jujudusud** [reply]
What about using a server with a little ping and no "enabled small network buffer" and 128 frames / buffer ?
This would avoid your internet fluctuations problems and make the jitter lower and the sound better.
